### PR TITLE
ClustersTable: add sorting

### PR DESCRIPTION
**Pull request by mareklibra**
Depends on:
- [x] #109

### CODE REVIEW

File: src/components/clusters/ClustersTable.tsx [hunk 1]
```diff
@@ -10,44 +21,80 @@ interface ClustersTableProps {
   deleteCluster: (id: string) => void;
 }
 
+// TODO(mlibra): let's try to remove it - not needed for HostsTable
```
Comment (mareklibra) [inline src/components/clusters/ClustersTable.tsx:24]: I will remove this extra formatting in a follow-up and let's discuss there.

Comment (mareklibra) [reply src/components/clusters/ClustersTable.tsx:24]: https://github.com/openshift-metal3/facet/pull/111

File: src/api/utils.ts [hunk 2]
```diff
@@ -25,3 +31,25 @@ export const handleApiError = <T>(error: AxiosError<T>, onError?: OnError) => {
     if (onError) return onError(error);
   }
 };
+
+type getCellType = (row: IRow, index: number | undefined) => string | HumanizedSortable;
+
+export const rowSorter = (sortBy: ISortBy, getCell: getCellType) => (a: IRow, b: IRow): number => {
```
Comment (jtomasek) [inline src/api/utils.ts:37]: Would it make more sense to move this into `components/ui/table/utils.ts`?

File: src/components/clusters/ClustersTable.tsx [hunk 3]
```diff
@@ -1,7 +1,18 @@
 import React from 'react';
-import { Table, TableHeader, TableBody, TableVariant, IRowData } from '@patternfly/react-table';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  IRowData,
+  sortable,
```
Comment (jtomasek) [inline src/components/clusters/ClustersTable.tsx:7]: this needs to be used from `src/components/ui/table/sortable`

File: src/api/utils.ts (hunk 2 above)
Comment (mareklibra) [reply src/api/utils.ts]: done

File: src/components/clusters/ClustersTable.tsx (hunk 3 above)
Comment (mareklibra) [reply src/components/clusters/ClustersTable.tsx]: done

File: src/components/clusterConfiguration/HostRowDetail.tsx [hunk 4]
```diff
@@ -32,7 +32,8 @@ type HostDetailItemProps = {
         value?: string;
       }[]
     | string
-    | number;
+    | number
+    | React.ReactNode;
```
Comment (jtomasek) [inline src/components/clusterConfiguration/HostRowDetail.tsx:36]: `ReactNode` already includes `string` and `number`

**Pull request merged**